### PR TITLE
task: update linter and rules

### PR DESCRIPTION
**Pull request by gssbzn**
Update to lastest golang linter and update some rules

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -29,6 +29,7 @@ jobs:
         golang:
           - 1.18
           - 1.19
+          - 1.20
```
Comment (gssbzn) [inline .github/workflows/pr.yml:32]: drive-by

File: .golangci.yml [hunk 2]
```diff
@@ -65,7 +65,7 @@ linters:
   disable-all: true
   enable:
     - bodyclose # checks whether HTTP response body is closed successfully [fast: false, auto-fix: false]
-    - deadcode # Finds unused code [fast: false, auto-fix: false]
```
Comment (gssbzn) [inline .golangci.yml:68]: remove deprecated rules

File: .golangci.yml [hunk 3]
```diff
@@ -92,20 +92,21 @@ linters:
     - revive # Fast, configurable, extensible, flexible, and beautiful linter for Go. Drop-in replacement of golint. [fast: false, auto-fix: false]
     - rowserrcheck # checks whether Err of rows is checked successfully [fast: false, auto-fix: false]
     - staticcheck  #megacheck): Staticcheck is a go vet on steroids, applying a ton of static analysis checks [fast: false, auto-fix: false]
-    - structcheck # Finds unused struct fields [fast: false, auto-fix: false]
     - stylecheck # Stylecheck is a replacement for golint [fast: false, auto-fix: false]
     - tagliatelle # Checks the struct tags. [fast: true, auto-fix: false]
     - thelper # thelper detects golang test helpers without t.Helper() call and checks the consistency of test helpers [fast: false, auto-fix: false]
+    - testableexamples # linter checks if examples are testable (have an expected output) [fast: true, auto-fix: false]
```
Comment (gssbzn) [inline .golangci.yml:98]: adding since we have examples

**Review by wtrocki — APPROVED** [review]
Thank you! Am I ok to follow up with the go-client lint config as well?
